src/order/ideal.lean

```diff
@@ -102,12 +113,33 @@ instance : order_bot (ideal P) :=
 
 end order_bot
 
+section order_top
+
+variables [order_top P]
+
 /-- There is a top ideal when `P` has a top element. -/
-instance {P} [order_top P] : order_top (ideal P) :=
+instance : order_top (ideal P) :=
 { top := principal ⊤,
   le_top := λ I x h, le_top,
   .. ideal.partial_order }
 
+@[simp] lemma top_carrier : (⊤ : ideal P).carrier = set.univ :=
+set.univ_subset_iff.1 (λ p _, le_top)
+
+lemma top_of_mem_top {I : ideal P} (topmem : ⊤ ∈ I) : I = ⊤ :=
+begin
+  ext,
+  change x ∈ I.carrier ↔ x ∈ (⊤ : ideal P).carrier,
+  split,
+  { simp [top_carrier] },
+  { exact λ _, I.mem_of_le le_top topmem }
+end
+
+lemma proper_of_ntop {I : ideal P} (ntop : I ≠ ⊤) : proper I :=
```

Comment:
I think this is the last thing left:
```suggestion
lemma proper_of_ne_top {I : ideal P} (ntop : I ≠ ⊤) : proper I :=
```

Once you accept this suggestion, feel free to put this PR on the merge queue by posting a comment with either `bors merge` or `bors r+` on its own line.